manager/manager.go

```diff
@@ -237,6 +237,12 @@ type manager struct {
 
 // Start the container manager.
 func (self *manager) Start() error {
+	//add-by fzu-huang clean the factories which is a external object.
+	//Usually we just run only one manager.So every time we start a manager
+	//we have to clear old handlerfactories, otherwise, new handlerfactories
+	//will be in behand of old factories. and won't be called.
+	container.ClearContainerHandlerFactories()
```

Comment:
I read the code roughly and I found that we init  fs data by `fs/fs.go: NewFsInfo()`.And we put this fsInfo into a new factory when we called manager.Start(). If we run two manager with a global factories array, the handlers of new manager's factory won't be called
```
// Register a ContainerHandlerFactory. These should be registered from least general to most general
// as they will be asked in order whether they can handle a particular container.
func RegisterContainerHandlerFactory(factory ContainerHandlerFactory, watchTypes []watcher.ContainerWatchSource) {
	factoriesLock.Lock()
	defer factoriesLock.Unlock()

	for _, watchType := range watchTypes {
		factories[watchType] = append(factories[watchType], factory)
	}
}
```

```
```
I think that